### PR TITLE
feat(cli): add 'hyperlane warp read' command

### PR DESCRIPTION
**Pull request by nbayindirli**
### Description

* adds new `hyperlane warp read` command

### Related issues

- fixes https://github.com/hyperlane-xyz/issues/issues/1189

### Backward compatibility

* yes

### Testing

* [x] manual
* [ ] ci-test

### CODE REVIEW

File: typescript/cli/src/commands/warp.ts [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 
 /**
  * Parent command
  */
 export const warpCommand: CommandModule = {
   command: 'warp',
   describe: 'Manage Hyperlane warp routes',
-  builder: (yargs) => yargs.command(config).version(false).demandCommand(),
+  builder: (yargs) =>
+    yargs.command(config).command(read).version(false).demandCommand(),
```
[the remark this answers is not in the document]
Comment (nbayindirli) [reply typescript/cli/src/commands/warp.ts:19]: this is how we have it set throughout the whole CLI

File: typescript/cli/src/commands/warp.ts [hunk 2]
```diff
@@ -39,3 +47,36 @@ export const config: CommandModuleWithContext<{
...
+  handler: async ({ context, chain, address, out }) => {
+    const { multiProvider } = context;
+    const evmERC20WarpRouteReader = new EvmERC20WarpRouteReader(
+      multiProvider,
+      chain,
+    );
+    const warpRouteConfig = await evmERC20WarpRouteReader.deriveWarpRouteConfig(
+      address,
+    );
+    writeFileAtPath(out, warpRouteConfig + '\n');
```
Comment (yorhodes) [inline typescript/cli/src/commands/warp.ts:78]: would be nice to log to stdout too and/or have a flag for this

**Review by yorhodes — APPROVED** [review]
have you tested this?

File: typescript/cli/src/commands/warp.ts (hunk 2 above)
Comment (nbayindirli) [reply typescript/cli/src/commands/warp.ts:78]: made `out` optional– so if present it writes, if not it's printed :)

**Comment by nbayindirli** [comment]
> have you tested this?

Yessir, here's the output with the new stdout logic:
```
➜  cli git:(noah/warp-read) hl warp read --chain ethereum --address 0x8b4192B9Ad1fCa440A5808641261e5289e6de95D                                       
Hyperlane CLI
(node:65034) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Hyperlane Warp Reader
---------------------
✅ Warp route config read successfully:
{
    "type": "collateral",
    "token": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
    "mailbox": "0xc005dc82818d67AF737725bD4bf75435d065D239",
    "owner": "0xa7ECcdb9Be08178f896c26b7BbD8C3D4E844d9Ba",
    "hook": {
        "address": "0x19b2cF952b70b217c90FC408714Fbc1acD29A6A8",
        "type": "aggregationHook",
        "hooks": [
            {
                "owner": "0xa7ECcdb9Be08178f896c26b7BbD8C3D4E844d9Ba",
                "address": "0x3A66Dc852e56d3748838b3C27CF381105b83705b",
                "paused": false,
                "type": "pausableHook"
            },
            {
                "address": "0x48e6c30B97748d1e2e03bf3e9FbE3890ca5f8CCA",
                "type": "merkleTreeHook"
            },
            {
                "owner": "0xa7ECcdb9Be08178f896c26b7BbD8C3D4E844d9Ba",
                "address": "0x9e6B1022bE9BBF5aFd152483DAD9b88911bC8611",
                "type": "interchainGasPaymaster",
                "beneficiary": "0xa7ECcdb9Be08178f896c26b7BbD8C3D4E844d9Ba",
                "oracleKey": "0xa7ECcdb9Be08178f896c26b7BbD8C3D4E844d9Ba",
                "overhead": {
                    "ancient8": 159736,
                    "arbitrum": 166887,
                    "gnosis": 160545,
                    "mantapacific": 182044,
                    "injective": 600000,
                    "avalanche": 159736,
                    "inevm": 159736,
                    "base": 160948,
                    "moonbeam": 160545,
                    "blast": 159736,
                    "bsc": 160545,
                    "celo": 160948,
                    "mode": 159337,
                    "neutron": 600000,
                    "polygonzkevm": 159736,
                    "scroll": 160545,
                    "polygon": 160545,
                    "optimism": 160948,
                    "redstone": 159337,
                    "viction": 159736,
                    "zetachain": 159337
                },
                "oracleConfig": {
                    "ancient8": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1632914711"
                    },
                    "arbitrum": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1579075739"
                    },
                    "gnosis": {
                        "tokenExchangeRate": "5162901",
                        "gasPrice": "4725948799906"
                    },
                    "mantapacific": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1475931288"
                    },
                    "injective": {
                        "tokenExchangeRate": "132855832",
                        "gasPrice": "72947441832"
                    },
                    "avalanche": {
                        "tokenExchangeRate": "170504166",
                        "gasPrice": "143654675730"
                    },
                    "inevm": {
                        "tokenExchangeRate": "110912472",
                        "gasPrice": "220838290191"
                    },
                    "base": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1623532813"
                    },
                    "moonbeam": {
                        "tokenExchangeRate": "1300455",
                        "gasPrice": "18762360700667"
                    },
                    "blast": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1632914711"
                    },
                    "bsc": {
                        "tokenExchangeRate": "2928726277",
                        "gasPrice": "8331132197"
                    },
                    "celo": {
                        "tokenExchangeRate": "3794136",
                        "gasPrice": "6418587052918"
                    },
                    "mode": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1636027075"
                    },
                    "neutron": {
                        "tokenExchangeRate": "398113300000000",
                        "gasPrice": "24343"
                    },
                    "polygonzkevm": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "3950000000"
                    },
                    "scroll": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1626640385"
                    },
                    "polygon": {
                        "tokenExchangeRate": "3406234",
                        "gasPrice": "7163220666867"
                    },
                    "optimism": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1623532813"
                    },
                    "redstone": {
                        "tokenExchangeRate": "15000000000",
                        "gasPrice": "1636027075"
                    },
                    "viction": {
                        "tokenExchangeRate": "2179365",
                        "gasPrice": "11238925410577"
                    },
                    "zetachain": {
                        "tokenExchangeRate": "7859938",
                        "gasPrice": "3122213704406"
                    }
                }
            }
        ]
    },
    "interchainSecurityModule": {
        "address": "0x9c764dF6bdd6B716CfadEAa348E1dB146694F67F",
        "type": "staticAggregationIsm",
        "modules": [
            {
                "address": "0x054B75b89DA7a428C2BF1f6b5df33944167B785C",
                "type": "merkleRootMultisigIsm",
                "validators": [
                    "0x95C7bF235837cb5A609fe6c95870410B9f68bCFF",
                    "0xa5a56e97fB46f0AC3A3D261E404ACB998d9a6969",
                    "0xBB5842aE0e05215b53Df4787A29144EfB7e67551"
                ],
                "threshold": 2
            },
            {
                "address": "0xD32513263Eb3190c4DA126fe418864904D9F4A32",
                "type": "messageIdMultisigIsm",
                "validators": [
                    "0x95C7bF235837cb5A609fe6c95870410B9f68bCFF",
                    "0xa5a56e97fB46f0AC3A3D261E404ACB998d9a6969",
                    "0xBB5842aE0e05215b53Df4787A29144EfB7e67551"
                ],
                "threshold": 2
            }
        ],
        "threshold": 1
    },
    "name": "USD Coin",
    "symbol": "USDC",
    "totalSupply": "24166384701944703",
    "decimals": 6
}
➜  cli git:(noah/warp-read) 
```

**Pull request merged**